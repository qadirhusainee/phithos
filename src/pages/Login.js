import React, { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import type { FunctionComponent } from "react";
import {
  View,
  Text,
  StyleSheet,
  Button,
  TextInput,
  Dimensions,
  SafeAreaView,
} from "react-native";

import { updateLoggedIn, updateUserDetail } from "../redux/actions/auth";
import Baner from "../component/Baner";

const { width, height } = Dimensions.get("window");

const emailPattern = /^\w+([-+.']\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*$/;
const passwordPattern = /^.{6,100}$/;

const Login: FunctionComponent = ({ navigation }) => {
  const dispatch = useDispatch();
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [error, setError] = useState("");

  const _onLogin = () => {
    let valid = false;
    if (
      email &&
      password &&
      emailPattern.test(email) &&
      passwordPattern.test(password) &&
      email.toLocaleLowerCase() === "qadirhusainee@gmail.com" &&
      password === "Password@123"
    ) {
      dispatch(updateLoggedIn(true));
      dispatch(updateUserDetail({ email, password }));
    } else {
      setError("Please enter valid credentials");
    }
  };
  return (
    <SafeAreaView style={{ flex: 1 }}>
      <Baner />
      <View style={{ flex: 1, alignItems: "center", marginTop: height / 4 }}>
        <TextInput
          style={styles.input}
          placeholder="Email"
          onChangeText={setEmail}
          value={email}
          keyboardType="email-address"
        />
        <TextInput
          style={styles.input}
          placeholder="password"
          value={password}
          onChangeText={setPassword}
        />
        {error ? <Text style={styles.error}>{error}</Text> : <View />}
        <Button
          style={styles.button}
          title="Login"
          onPress={() => {
            _onLogin();
          }}
        />
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {},
  input: {
    borderWidth: 1,
    borderColor: "black",
    borderRadius: 5,
    width: width - 100,
    height: 50,
    padding: 10,
    marginVertical: 10,
  },
  error: {
    color: "red",
  },
  button: {
    marginTop: 20,
  },
});

export default Login;
