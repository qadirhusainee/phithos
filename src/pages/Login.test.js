import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from "react-native-testing-library";

import Login from "./Login";

jest.mock("react-redux", () => {
  const ActualReactRedux = require.requireActual("react-redux");
  return {
    ...ActualReactRedux,
    useSelector: jest.fn().mockImplementation(() => {
      return { showBaner: true };
    }),
    useDispatch: jest.fn().mockImplementation(() => jest.fn()),
  };
});

describe("Login", () => {
  it("Matches Snapshot", () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Showing Baner", () => {
    const { queryByText } = render(<Login />);

    const element = queryByText("Emulator");
    expect(element).toBeDefined();
  });

  it("Verifying entered input values", () => {
    const dispatch = jest.fn();
    const { getAllByA11yLabel, queryByText, queryByDisplayValue } = render(
      <Login />
    );

    const email = getAllByA11yLabel("email");
    const password = getAllByA11yLabel("password");

    fireEvent.changeText(email[0], "qadirhusainee@gmail.com");
    fireEvent.changeText(password[0], "Password@123");
    const inputEmail = queryByDisplayValue("qadirhusainee@gmail.com");
    const inpupassword = queryByDisplayValue("Password@123");
    fireEvent.press(queryByText("Login"));
    expect(inputEmail).toBeDefined();
    expect(inpupassword).toBeDefined();
  });
});
