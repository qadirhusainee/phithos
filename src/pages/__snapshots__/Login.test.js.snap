// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Matches Snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2f2f2",
        "borderColor": "grey",
        "borderWidth": 0.5,
        "height": 80,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "paddingTop": 5,
        }
      }
    >
      Simulator(IOS)
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "marginRight": 20,
          "width": 750,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Close
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 333.5,
      }
    }
  >
    <TextInput
      accessibilityLabel="email"
      allowFontScaling={true}
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="Email"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 50,
          "marginVertical": 10,
          "padding": 10,
          "width": 650,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      accessibilityLabel="password"
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="password"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 50,
          "marginVertical": 10,
          "padding": 10,
          "width": 650,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View />
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
